pykafka/rdkafka/producer.py

```diff
@@ -0,0 +1,163 @@
+from concurrent import futures
+import logging
+import weakref
+
+from pykafka.exceptions import (
+        KafkaException, RdKafkaStoppedException, ProducerStoppedException)
+from pykafka.producer import Producer, CompressionType, random_partitioner
+from pykafka.utils.compat import get_bytes
+from . import _rd_kafka
+
+
+log = logging.getLogger(__name__)
+
+
+class RdKafkaProducer(Producer):
+    """A librdkafka-backed version of pykafka.Producer
+
+    This aims to conform to the Producer interface as closely as possible.
+    For an overview of how configuration keys are mapped to librdkafka's, see
+    _mk_rdkafka_config_lists.
+    """
+    def __init__(self,
+                 cluster,
+                 topic,
+                 partitioner=random_partitioner,
+                 compression=CompressionType.NONE,
+                 max_retries=3,
+                 retry_backoff_ms=100,
+                 required_acks=1,
+                 ack_timeout_ms=10 * 1000,
+                 max_queued_messages=100000,
+                 min_queued_messages=2000,  # NB differs from pykafka.Producer
+                 linger_ms=5 * 1000,
+                 block_on_queue_full=True,
+                 sync=False):
+        callargs = {k: v for k, v in vars().items()
+                         if k not in ("self", "__class__")}
+        self._rdk_producer = None
+        # super() must come last because it calls start()
+        super(RdKafkaProducer, self).__init__(**callargs)
+
+    def start(self):
+        if not self._running:
+            brokers = b','.join(b.host + b":" + get_bytes(str(b.port))
+                               for b in self._cluster.brokers.values())
+            conf, topic_conf = self._mk_rdkafka_config_lists()
+
+            self._rdk_producer = _rd_kafka.Producer()
+            self._rdk_producer.configure(conf=conf)
+            self._rdk_producer.configure(topic_conf=topic_conf)
```

Comment:
An additional comment here about the order of these `configure` calls mattering would be helpful.